course/test/src/Ant_TEST.cc

```diff
@@ -0,0 +1,129 @@
+#include <cmath>
+#include <sstream>
+#include <string>
+#include <memory>
+
+#include "gtest/gtest.h"
+
+#include "ant.h"
+#include "cell_mock.h"
+
+namespace ekumen {
+namespace simulation {
+namespace test {
+namespace {
+
+GTEST_TEST(AntTest, OccupiedCellOnInitialization) {
+  std::shared_ptr<Cell> cell(new Cell());
+  Ant ant(cell);
+  EXPECT_FALSE(cell->IsFree());
+}
+
+GTEST_TEST(AntTest, CallbackAssignedCorrectly) {
+  std::shared_ptr<Cell> cell(new Cell());
+  Ant ant = Ant(cell);
+  bool lambda_called = false;
+  auto callback = [&](InsectCallbackMetrics& metrics) {
+        lambda_called = true;
+  };
+  ant.SetRoundResultsCallback(callback);
+  ant.RunRound();
+
+  EXPECT_TRUE(lambda_called);
+}
+
+GTEST_TEST(AntTest, IsAliveOnInit) {
+  std::shared_ptr<Cell> cell(new Cell());
+  Ant ant = Ant(cell);
+  InsectCallbackMetrics test_metrics;
+  auto callback = [&](InsectCallbackMetrics& metrics) {
+        if (metrics.IsDead()) {
+          test_metrics.SetDead();
+        }
+  };
+
+  ant.SetRoundResultsCallback(callback);
+  ant.RunRound();
+  EXPECT_FALSE(test_metrics.IsDead());
+}
+
+GTEST_TEST(AntTest, IsDeadWhenToldToDie) {
+  std::shared_ptr<Cell> cell(new Cell());
+  Ant ant = Ant(cell);
+  InsectCallbackMetrics test_metrics;
+  auto callback = [&](InsectCallbackMetrics& metrics) {
+        if (metrics.IsDead()) {
+          test_metrics.SetDead();
+        }
+  };
+
+  ant.SetRoundResultsCallback(callback);
+  ant.RunRound();
+  EXPECT_FALSE(test_metrics.IsDead());
+  ant.Die();
+  ant.RunRound();
+  EXPECT_TRUE(test_metrics.IsDead());
+}
+
+GTEST_TEST(AntTest, AntMovedToOtherCell) {
+  std::shared_ptr<Cell> free_cell(new Cell());
+  SurroundingCells surr({free_cell, free_cell, free_cell, free_cell});
+  std::shared_ptr<Cell> center_cell(new Cell());
+  center_cell->SetSurroundingCells(surr);
+  std::shared_ptr<Cell> center_cell_ptr2 = center_cell;
```

Comment:
Why do you need a new variable pointing to the same place?

Reply:
We don't. Addressed.